### PR TITLE
fix error checking in kl.killPod within SyncPod

### PR DESCRIPTION
**Pull request by zhifei92**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:
fix error checking in kl.killPod within SyncPod: when `kl.killPod` returns nil, `wait.Interrupted ` always returns `false, nil`, and SyncPod consequently returns — this is not the expected behavior. move it to the correct position
```
    if err := kl.killPod(ctx, pod, p, nil); err == nil {
        if wait.Interrupted(err) {
            return false, err
        }
```

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
refs #117412, hi @smarterclayton , could you review it for me, thx.
#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by bart0sh** [comment]
/triage accepted
/priority important-longterm
/ok-to-test
/release-note-none

@zhifei92 Thank you for your contribution! As this is a bugfix, please provide e2e test case to cover the issue.

File: pkg/kubelet/kubelet.go
Comment (bart0sh) [inline pkg/kubelet/kubelet.go:1868]:
```suggestion
				if wait.Interrupted(err) {
				        klog.ErrorS(err, "KillPod interrupted", "pod", klog.KObj(pod), "podStatus", podStatus)
```

Comment (zhifei92) [reply pkg/kubelet/kubelet.go:1868]: >Any early interruption should result in exiting the Sync*Pod method as quickly as possible without logging intermediate errors.

should we maintain consistency with other places in Sync*Pod where `wait.Interrupted` is called?

Comment (bart0sh) [reply pkg/kubelet/kubelet.go:1868]: agreed.

Comment (bart0sh) [reply pkg/kubelet/kubelet.go:1868]: /lgtm

**Comment by zhifei92** [comment]
@derekwaynecarr hi, this PR needs an approval label, please help review it.

**Comment by zhifei92** [comment]
hi @rphillips @bart0sh do we need anyone's else review?

**Comment by bart0sh** [comment]
> do we need anyone's else review?

As this pr already lgtm-ed, we need SIG-node approvers to take a look

/assign @mrunlap @derekwaynecarr @dchen1107 
for approval

**Comment by dims** [comment]
poke @kubernetes/sig-node-leads for approval!

**Review by mrunalp — APPROVED** [review]

**Pull request merged**
